## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 493213`
Commit: [`5491524016e88ca00da4b060abc71b788f64c883`](https://github.com/NixOS/nixpkgs/commit/5491524016e88ca00da4b060abc71b788f64c883) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/5491524016e88ca00da4b060abc71b788f64c883..pull/493213/head))
Merge: [`1b2a380a21e347165a212157b5439d4273546f51`](https://github.com/NixOS/nixpkgs/commit/1b2a380a21e347165a212157b5439d4273546f51)

Logs: https://github.com/JamieMagee/nixpkgs-review-gha/actions/runs/22293178147


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.gentex_homelink</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.homelink-integration-api</li>
    <li>python313Packages.homelink-integration-api.dist</li>
    <li>python314Packages.homelink-integration-api</li>
    <li>python314Packages.homelink-integration-api.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.gentex_homelink</summary>
<pre>no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonCatchConflictsPhase&quot; }
Running phase: pythonRemoveBinBytecodePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRemoveBinBytecodePhase&quot; }
Running phase: pythonImportsCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pytestCheckPhase&quot; }
Executing pytestCheckPhase
pytest flags: -m pytest tests/components/gentex_homelink --dist=loadfile --showlocals
ImportError while loading conftest &#x27;/build/source/tests/components/gentex_homelink/conftest.py&#x27;.
tests/components/gentex_homelink/conftest.py:10: in &lt;module&gt;
    from homeassistant.components.gentex_homelink import DOMAIN
homeassistant/components/gentex_homelink/__init__.py:5: in &lt;module&gt;
    from homelink.mqtt_provider import MQTTProvider
/nix/store/79k869ba3l960yivfiq23qr443rj0cnx-python3.13-homelink-integration-api-0.0.1/lib/python3.13/site-packages/homelink/mqtt_provider.py:7: in &lt;module&gt;
    import paho.mqtt.client as mqtt
E   ModuleNotFoundError: No module named &#x27;paho&#x27;</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.gentex_homelink</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.homelink-integration-api</li>
    <li>python313Packages.homelink-integration-api.dist</li>
    <li>python314Packages.homelink-integration-api</li>
    <li>python314Packages.homelink-integration-api.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.gentex_homelink</summary>
<pre>no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonCatchConflictsPhase&quot; }
Running phase: pythonRemoveBinBytecodePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRemoveBinBytecodePhase&quot; }
Running phase: pythonImportsCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonImportsCheckPhase&quot; }
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pytestCheckPhase&quot; }
Executing pytestCheckPhase
pytest flags: -m pytest tests/components/gentex_homelink --dist=loadfile --showlocals
ImportError while loading conftest &#x27;/build/source/tests/components/gentex_homelink/conftest.py&#x27;.
tests/components/gentex_homelink/conftest.py:10: in &lt;module&gt;
    from homeassistant.components.gentex_homelink import DOMAIN
homeassistant/components/gentex_homelink/__init__.py:5: in &lt;module&gt;
    from homelink.mqtt_provider import MQTTProvider
/nix/store/69f2a5jy1667kbnc3lxj32glh8qvw4la-python3.13-homelink-integration-api-0.0.1/lib/python3.13/site-packages/homelink/mqtt_provider.py:7: in &lt;module&gt;
    import paho.mqtt.client as mqtt
E   ModuleNotFoundError: No module named &#x27;paho&#x27;</pre>
</details>
</details>

--- a/pkgs/development/python-modules/homelink-integration-api/default.nix
+++ b/pkgs/development/python-modules/homelink-integration-api/default.nix
@@ -1,0 +1,49 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  aiofiles,
+  aiohttp,
+  boto3,
+  paho-mqtt,
+  pyopenssl,
+  python-decouple,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "homelink-integration-api";
+  version = "0.0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Gentex-Corporation";
+    repo = "homelink-integration-api";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-ELEqx41JSAmXBEowwJ1tYPZV40hMjswaHQonD+1IG5E=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    aiofiles
+    aiohttp
+    boto3
+    paho-mqtt
+    pyopenssl
+    python-decouple
+  ];
+
+  # upstream tests require network access and AWS credentials
+  doCheck = false;
+
+  pythonImportsCheck = [ "homelink" ];
+
+  meta = {
+    description = "API to interact with Homelink cloud for MQTT-enabled smart home platforms";
+    homepage = "https://github.com/Gentex-Corporation/homelink-integration-api";
+    changelog = "https://github.com/Gentex-Corporation/homelink-integration-api/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -2060,7 +2060,8 @@
       ];
     "gentex_homelink" =
       ps: with ps; [
-      ]; # missing inputs: homelink-integration-api
+        homelink-integration-api
+      ];
     "geo_json_events" =
       ps: with ps; [
         aio-geojson-generic-client
@@ -7449,6 +7450,7 @@
     "generic_hygrostat"
     "generic_thermostat"
     "geniushub"
+    "gentex_homelink"
     "geo_json_events"
     "geo_location"
     "geo_rss_events"

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7011,6 +7011,8 @@ self: super: with self; {
 
   homeconnect = callPackage ../development/python-modules/homeconnect { };
 
+  homelink-integration-api = callPackage ../development/python-modules/homelink-integration-api { };
+
   homematicip = callPackage ../development/python-modules/homematicip { };
 
   homepluscontrol = callPackage ../development/python-modules/homepluscontrol { };

